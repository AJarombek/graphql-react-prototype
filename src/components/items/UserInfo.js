/**
 * UserInfo component which displays general information about the user that the dashboard is for.
 * @author Andrew Jarombek
 * @since 4/11/2020
 */

import React, { useEffect, useState } from 'react';
import { getUserInfo } from '../../datasource/GraphQL';

const UserInfo = () => {
  const [user, setUser] = useState({});
  const [error, setError] = useState(null);

  useEffect(() => {
    async function getGraphQLResult() {
      const result = await getUserInfo('AJarombek');

      console.info(result);
      if (result.data.data) {
        setUser(result.data.data.user);
        setError(null);
      } else {
        setError(result.data.errors[0].message);
      }
    }

    getGraphQLResult()
  }, []);

  return (
    <div className="items user-info">
      {error ?
        <div className="error">
          <h6>{error}</h6>
        </div>
        :
        <>
          <figure>
            <img src={user.avatarUrl} alt=""/>
          </figure>
          <h3>{user.name}</h3>
          <p>{user.location}</p>
          <a href={`https:\\\\${user.websiteUrl}`}>Personal Website</a>
        </>
      }
    </div>
  );
};

export default UserInfo;
