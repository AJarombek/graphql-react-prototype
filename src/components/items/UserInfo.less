/**
 * Stylesheet for the UserInfo component.
 * Author: Andrew Jarombek
 * Date: 4/11/2020
 */

@import "../../vars";

.user-info {
  grid-area: userInfo;

  display: grid;
  grid-template-columns: 1fr 3fr;
  grid-template-areas:
    "logo name    "
    "logo location"
    "logo website ";

  height: 125px;

  figure {
    grid-area: logo;
    height: 100px;
    width: 100px;
    margin: 0;

    img {
      height: 100%;
      width: 100%;
      border-radius: 50%;
    }
  }

  h3, p, a {
    margin: 0 0 0 10px;
  }

  h3 {
    grid-area: name;
  }

  p {
    grid-area: location;
  }

  a {
    grid-area: website;
    transition: color ease 0.4s;

    &:hover {
      color: @contrastPurple;
    }
  }

  .error {
    grid-column: 1 e("/") 3;
    grid-row: 1 e("/") 4;
  }
}
