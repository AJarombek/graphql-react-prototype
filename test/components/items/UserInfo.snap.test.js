/**
 * Snapshot test for the UserInfo component.
 * @author Andrew Jarombek
 * @since 4/25/2020
 */

import React from 'react';
import UserInfo from '../../../src/components/items/UserInfo';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<UserInfo />).toJSON();
  expect(tree).toMatchSnapshot();
});
