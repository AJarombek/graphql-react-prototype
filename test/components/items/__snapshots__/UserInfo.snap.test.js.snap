// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="items user-info"
>
  <figure>
    <img
      alt=""
    />
  </figure>
  <h3 />
  <p />
  <a
    href="https:\\\\\\\\undefined"
  >
    Personal Website
  </a>
</div>
`;
